xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/DiagramMacros.xml

```diff
@@ -0,0 +1,446 @@
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </cache>
+      <code>
+        <contenttype>PureText</contenttype>
+        <disabled>0</disabled>
+        <editor>PureText</editor>
+        <name>code</name>
+        <number>2</number>
+        <prettyName>Code</prettyName>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require(['jquery', 'xwiki-events-bridge'], function($) {
+  var serviceURL = new XWiki.Document(
+                       XWiki.Model.resolve('Confluence.Macros.DiagramService', XWiki.EntityType.DOCUMENT))
+                       .getURL('get', 'outputSyntax=plain'),
+      createSuccessMessage = "$escapetool.javascript($services.localization.render('confluencediagram.create.success'))",
+      createFailureMessage = "$escapetool.javascript($services.localization.render('confluencediagram.create.error'))",
+      deleteSuccessMessage = "$escapetool.javascript($services.localization.render('confluencediagram.delete.success'))",
+      deleteFailureMessage = "$escapetool.javascript($services.localization.render('confluencediagram.delete.error'))";
+
+  var handleCreateDiagram = function () {
+    $('.create-diagram-btn').on('click', function(e){
+      e.preventDefault();
+
+      var pageReferenceSerialized = $(this).prevAll('.create-diagram-page:first').val(),
+          diagramName = $(this).prevAll('.create-diagram-name:first').val();
```

Comment:
No need for ``:first``, ``val()`` returns the value of the first item in the selected list.

Reply:
Fixed